test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -48,6 +48,44 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Tests
 
+(deftest ^:integration symlink-cadir-test
+  (testing "does not symlink if custom cadir"
+    (let [tmpdir (ks/temp-dir)
+          cadir (str tmpdir "/foo/bar")
+          confdir (str tmpdir "/puppet")
+          ssldir (str confdir "/ssl")
+          content (str "[main]"
+                       "\n  cadir = " cadir
+                       "\n  ssldir = " ssldir
+                       "\n  confdir = " confdir)
+          config (str tmpdir "/puppet.conf")]
+      (spit config cadir)
+      (testutils/with-puppet-conf-files
+        {"puppet.conf" config} bootstrap/master-conf-dir
+        (ca/symlink-cadir cadir)
+        (is (not (fs/exists? (str ssldir "/ca"))))
+        (is (not (fs/exists? cadir))))))
+  (testing "symlinks correctly and remotes existing old-cadir if needed"
+    (let [tmpdir (ks/temp-dir)
+          cadir (str tmpdir "/puppetserver/ca")
+          confdir (str tmpdir "/puppet")
+          ssldir (str confdir "/ssl")
+          content (str "[main]"
+                       "\n  cadir = " cadir
+                       "\n  ssldir = " ssldir
+                       "\n  confdir = " confdir)
+          config (str tmpdir "/puppet.conf")
+          old-cadir (str ssldir "/ca")]
+      (spit config cadir)
+      (fs/mkdirs ssldir)
+      (fs/mkdirs cadir)
+      (testutils/with-puppet-conf-files
+        {"puppet.conf" config} bootstrap/master-conf-dir
+        (ca/symlink-cadir cadir)
+        (is (fs/exists? old-cadir))
```

Comment:
I think we should test that it's a symlink, rather than just existing.

Reply:
The assertion below this one resolves it as a symlink and ensures it's pointing to the new cadir, this is more of a sanity check that it even exists before hand.

Reply:
¯\_(ツ)_/¯ I feel like intentions are clearer and it's potentially easier to debug if this tests that it's a symlink. Like, if this exists as a regular directory, things will error when we try to `read-sym-link`, instead of giving us a nice test failure that it's not a symlink in the first place.